src/main/java/TaskManager.java

```diff
@@ -0,0 +1,215 @@
+import Exceptions.InvalidDeadlineFormatException;
+import Exceptions.InvalidEventFormatException;
+import Exceptions.InvalidTodoFormatException;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class TaskManager {
+    // private static final String FILE_PATH = "./data/tasks.txt";
+    private static final String FILE_PATH = "." + File.separator + "data" + File.separator + "tasks.txt";
+
+    public static final int DEADLINE_BEGIN_INDEX = 8;
+    public static final int EVENT_BEGIN_INDEX = 5;
+    public static final int TODO_BEGIN_INDEX = 4;
+    public static final int EVENT_MAX_PARTS = 3;
+    public static final int DEADLINE_MAX_PARTS = 2;
+    public static final int INDEX_OFFSET = 1;
+    public static final int PART_0 = 0;
+    public static final int PART_1 = 1;
+    public static final int PART_2 = 2;
+    public static final int START_INDEX = 0;
+    private ArrayList<Task> taskList = new ArrayList<>();
+    private int index = 0;
+    UserInterface userInterface = new UserInterface();
+
+    public void addTask(String taskDescription) {
+        String taskType = taskDescription.split(" ")[START_INDEX];
+        try {
+            switch (taskType) {
+                case "deadline":
+                    addDeadlineTask(taskDescription);
+                    break;
+                case "event":
+                    addEventTask(taskDescription);
+                    break;
+                case "todo":
+                    addTodoTask(taskDescription);
+                    break;
+                default:
+                    throw new NoSuchMethodException();
+            }
+        } catch (NoSuchMethodException e) {
+            userInterface.printInvalidTaskType(taskDescription);
+        } catch (InvalidDeadlineFormatException e) {
+            userInterface.printInvalidDeadlineFormat(e);
+        } catch (InvalidTodoFormatException e) {
+            userInterface.printInvalidTodoFormat(e);
+        } catch (InvalidEventFormatException e) {
+            userInterface.printInvalidEventFormat(e);
+        }
+    }
+
+    private void addDeadlineTask(String taskDescription) throws InvalidDeadlineFormatException {
```

Comment:
Can make use of more OOP by using a parser to parse inputs and command subclass such as "AddCommand" to execute the adding of the task to the task list. Can take reference from personsbook for implementation